examples/playbooks/increase_ovirt_node_cpus.yml

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright (c) 2018 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+---
+
+#
+# This example shows how to increase the number of CPUs of an OpenShift node
+# that is running as a virtual machine inside an oVirt environment.
+#
+# The playbook expects the IP address of the node inside the `instance`
+# variable. That can be achieved using the `extraVars` field and templating
+# mechanism of the auto-heal service. For example, the healing action could be
+# defined as follows:
+#
+#   awxJob:
+#     template: "Increase node CPUS"
+#     extraVars: |-
+#       {
+#         "instance": "{{ $labels.instance }}"
+#       }
+#
+# To use it the credentials for the oVirt environment have to be available in a
+# credential inside the AWX server.
+#
+
+- name: Increase node CPUs
+  hosts: localhost
+  connection: local
+  gather_facts: false
+
+  # These variables are intended to be overriden using the `--extra-vars`
+  # command line option:
+  vars:
+    # Maximum number of CPUs to assing to the VM.
+    max_cpus: 4
+    # Amount of CPUs to add in each increase.
+    inc_cpus: 1
+
+  tasks:
+
+  # Calculate the IP address from the `instance` input variable, and then do a
+  # reverse DNS lookup to find the FQDN, which in this environment is equal to
+  # the name of the virtual machine.
```

Comment:
Worth to mention another requirement for this hostname == k8s node name (returned by the alert)

Reply:
Note that the k8s node name is not involved at all here. The requirements are that the `instance` label of the alert contains the IP address of the node, and that the name of the virtual machine in oVirt is equal to the FQDN. The first is guaranteed by the way that the node exporter works. The second isn't guaranteed at all. I will try to add both things to the comments.

Reply:
Yes, when using the node exporter the `instance` label is the IP address and port of the metrics endpoint of the node for example `192.168.100.6:9100`.

Reply:
Note that this for the metrics that were used in this example, other metrics, from other sources, may have different `instance` labels.